### PR TITLE
Chore/ ynETH/ynEigen verify implementation and proxyadmins

### PR DESCRIPTION
**Pull request by danoctavian**
Add assertions to ensure the json deployment contracts never go out of sync, and the verify scripts always cross check all entries.

The focus is to see given the Proxy address, do the ProxyAdmin and implementation of said proxy match what's stored in the json.

Additionally checks are added for all withdrawals contracts.

Consolidates:

all deployed contracts into the single ynETH-1.json and ynETH-17000.json:

 delete deployments/ynETHWithdrawals-17000.json
 delete deployments/ynETHWithdrawalsProcessor-17000.json
 
 And add the contents to ynETH-17000.json
 
 add the missing ynViewer which acts as a rate provider to ynETH-1.json and ynETH-17000.json
 
 
 The following all pass:
 
 ```make ynEigen-verify  path=script/ynEigen/input/lsd-mainnet.json rpc="https://black-young-pool.quiknode.pro/[mykey]"```
 
 ```make ynEigen-verify  path=script/ynEigen/input/lsd-holesky.json rpc="https://rpc.ankr.com/eth_holesky"```
 
 ```forge script script/ynETH/VerifyYnETH.s.sol --rpc-url https://rpc.ankr.com/eth_holesky```
 
 ```forge script script/ynETH/VerifyYnETH.s.sol --rpc-url https://black-young-pool.quiknode.pro/[mykey]```

### CODE REVIEW

File: deployments/YnLSDe-1.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
   "implementation-eigenStrategyManager": "0xE2343a850394ee1b1e33b16Da5bD73B076D6Fa81",
   "implementation-rateProvider": "0xAEDC83F607126b4D2437F79285e2C635685BBA01",
   "implementation-tokenStakingNodesManager": "0x17e77F964538fed3ADEc3E8dBB6147dA258206Da",
-  "implementation-ynEigenDepositAdapter": "0x5F8c55A5A33d63CAa231c668a62f69AA862020E9",
+  "implementation-ynEigenDepositAdapter": "0xA9CD5a5199177b704287e379e85af95F0f061Bf2",
```
Comment (danoctavian) [inline deployments/YnLSDe-1.json:19]: implementation was not updated properly in the json file

File: deployments/ynETH-1.json [hunk 2]
```diff
@@ -17,15 +17,18 @@
   "implementation-rewardsDistributor": "0x306EBed66642867B94b29752be37b8f20fEb5F72",
   "implementation-stakingNodesManager": "0xaBD3A755E4eEF24f862c268D21Fd5235904cc811",
   "implementation-ynETH": "0x14DC3d915107DcA9ed39E29E14fbdfe4358a1346",
+  "implementation-ynViewer": "0x0365a6eF790e05EEe386B57326e5Ceaf5B10899e",
```
Comment (danoctavian) [inline deployments/ynETH-1.json:20]: added in ynViewer which was missing

File: deployments/ynETH-17000.json [hunk 3]
```diff
@@ -11,20 +11,32 @@
   "STAKING_NODES_OPERATOR": "0x9Dd8F69b62ddFd990241530F47dcEd0Dad7f7d39",
   "STAKING_NODE_CREATOR": "0x9Dd8F69b62ddFd990241530F47dcEd0Dad7f7d39",
   "VALIDATOR_MANAGER": "0x9Dd8F69b62ddFd990241530F47dcEd0Dad7f7d39",
-  "implementation-StakingNodesManager": "0x104f0c5156BC0D9Cbda3f30d1B74F42b5610A2dF",
-  "implementation-ynETH": "0x0E60A5db38c545ced3d3480B6768811cc603649A",
+  "implementation-stakingNodesManager": "0xC9cf6740282617f3B392f900De0449E687Ce05e3",
```
Comment (danoctavian) [inline deployments/ynETH-17000.json:14]: this was outdated post M3 upgrade

File: deployments/ynETH-17000.json [hunk 4]
```diff
@@ -11,20 +11,32 @@
...
+  "implementation-ynETH": "0x090D67d3C97712f6C17a037515CbB8502561EE57",
```
Comment (danoctavian) [inline deployments/ynETH-17000.json:15]: this was outdated post M3 upgrade

File: deployments/ynETH-17000.json [hunk 5]
```diff
@@ -11,20 +11,32 @@
...
   "implementation-executionLayerReceiver": "0x1fbedf3773418f20b9dfeafcd9d263030eb0e42f",
   "implementation-consensusLayerReceiver": "0xe7acc0533c650ad0cc11f57f81c38fa19634b1d7",
-  "implementation-RewardsDistributor": "0xb6ec4d9f71e437c672147c576f1c70ba5da8d159",
+  "implementation-rewardsDistributor": "0xb6ec4d9f71e437c672147c576f1c70ba5da8d159",
+  "implementation-withdrawalsProcessor": "0x36a38AA91947DbE6539e19512E6FF26576015Bb2",
```
Comment (danoctavian) [inline deployments/ynETH-17000.json:19]: add in the missing withdrawals extra contracts

File: script/Actors.sol [hunk 6]
```diff
@@ -64,7 +71,12 @@ contract ActorAddresses {
                 // TODO: replace with concrete deployment
                 YNStrategyController: 0x1234567890123456789012345678901234567890,
                 // TODO: replace with concrete deployment
-                YNTokenStakingNodeOperator: 0x2234567890123456789012345678901234567890
+                YNTokenStakingNodeOperator: 0x2234567890123456789012345678901234567890,
+                ynETHWithdrawalsWallets: YnETHWithdrawalsWallets({
```
Comment (danoctavian) [inline script/Actors.sol:75]: add in freshly created multisigs to handle withdrawals on Holesky

File: script/Actors.sol [hunk 7]
```diff
@@ -110,7 +122,14 @@ contract ActorAddresses {
             // TODO: replace with concrete deployment
             YNStrategyController: 0x0573A7DaFBc080064663623979287286Bb65C1BD,
             // TODO: replace with concrete deployment
-            YNTokenStakingNodeOperator: 0xfcad670592a3b24869C0b51a6c6FDED4F95D6975 // same as YNSecurityCouncil
+            YNTokenStakingNodeOperator: 0xfcad670592a3b24869C0b51a6c6FDED4F95D6975, // same as YNSecurityCouncil
+
+            // TODO: MUST fill for prod. same as YnDev for now
+            ynETHWithdrawalsWallets: YnETHWithdrawalsWallets({
+                YNStakingNodesWithdrawer: 0xa08F39d30dc865CC11a49b6e5cBd27630D6141C3,
```
Comment (danoctavian) [inline script/Actors.sol:129]: have YnDEV for now until set to newly minted multisigs

File: script/BaseScript.s.sol [hunk 8]
```diff
@@ -11,11 +11,20 @@ import {Script} from "lib/forge-std/src/Script.sol";
...
 import {ContractAddresses} from "script/ContractAddresses.sol";
+import {TransparentUpgradeableProxy} from "lib/openzeppelin-contracts/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import {ProxyAdmin} from "lib/openzeppelin-contracts/contracts/proxy/transparent/ProxyAdmin.sol";
 import {console} from "lib/forge-std/src/console.sol";
 
+
 abstract contract BaseScript is Script, Utils {
     using stdJson for string;
 
+    struct ProxyAddresses {
```
Comment (danoctavian) [inline script/BaseScript.s.sol:22]: load up all data from JSON file into this struct.

This way you can assert that the JSON matches the on-chain for the trinity:

Proxy - ProxyAdmin - Implementation

File: script/ynETH/BaseYnETHScript.s.sol [hunk 9]
```diff
@@ -65,13 +98,41 @@ abstract contract BaseYnETHScript is BaseScript {
...
+        proxies.consensusLayerReceiver = loadProxyAddresses(jsonContent, "consensusLayerReceiver");
+
         deployment.rewardsDistributor = RewardsDistributor(payable(jsonContent.readAddress(".proxy-rewardsDistributor")));
-        deployment.stakingNodeImplementation = StakingNode(payable(jsonContent.readAddress(".stakingNodeImplementation")));
+        proxies.rewardsDistributor = loadProxyAddresses(jsonContent, "rewardsDistributor");
+
+        deployment.ynViewer = ynViewer(payable(jsonContent.readAddress(".proxy-ynViewer")));
+        proxies.ynViewer = loadProxyAddresses(jsonContent, "ynViewer");
+
+        if (block.chainid == 17000) { // Holesky chain ID
```
Comment (danoctavian) [inline script/ynETH/BaseYnETHScript.s.sol:121]: nothing to parse until we deploy those

File: script/ynETH/VerifyYnETH.s.sol [hunk 10]
```diff
@@ -0,0 +1,717 @@
+/// SPDX-License-Identifier: BSD 3-Clause License
+pragma solidity ^0.8.24;
```
Comment (danoctavian) [inline script/ynETH/VerifyYnETH.s.sol:2]: the verify ynETH is now ehances to check:


Proxy - ProxyAdmin - Implementation Trinity matches the JSON with what's on-chain.

All the deps and fields of withdrawals are asserted

File: script/ynEigen/YnEigenVerifier.s.sol [hunk 11]
```diff
@@ -78,6 +79,84 @@ contract YnEigenVerifier is BaseYnEigenScript {
...
+        // Verify Implementation address
+        address implementationAddress = Utils.getTransparentUpgradeableProxyImplementationAddress(contractAddress);
+        require(
+            implementationAddress == proxyAddresses.implementation,
+            string.concat(contractName, ": Implementation address mismatch, expected: ", vm.toString(proxyAddresses.implementation), ", got: ", vm.toString(implementationAddress))
+        );
+        console.log(string.concat("\u2705 ", contractName, ": Implementation address - ", vm.toString(implementationAddress)));
+    }
+
+    function verifyProxies() internal view {
```
Comment (danoctavian) [inline script/ynEigen/YnEigenVerifier.s.sol:122]: check the trinity 

Proxy - ProxyAdmin - Implementation

for ynEigen as well

**Comment by danoctavian** [comment]
Screenshot 2024-10-10 at 11 44 06 PM
Screenshot 2024-10-10 at 11 44 46 PM
Screenshot 2024-10-10 at 11 46 00 PM
Screenshot 2024-10-10 at 11 46 32 PM

**Review by xhad — APPROVED** [review]

**Pull request merged**